### PR TITLE
schildichat-{web,desktop}: remove due to lack of maintenance

### PR DESCRIPTION
**Pull request by yu-re-ka**
## Description of changes

- Electron: The latest upstream schildichat-desktop release still uses Electron 24, while Electron 25 just went EOL: https://github.com/SchildiChat/schildichat-desktop/issues/215
- Apparently the flatpak version is still vulnerable to the libwebp CVE: https://github.com/SchildiChat/schildichat-desktop/issues/212
- Element-desktop [v1.11.36](https://github.com/vector-im/element-desktop/releases/tag/v1.11.36) released on Jul 18 has a fix for [CVE-2023-37259](https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword=CVE-2023-37259) which also effects schildichat, but has not been addressed/rebased since then.

Alternative to #267618

When this is done, I will make a new PR that sets the insecure flag targeted to 23.05/23.11 stable channels, as a compromise between "no breaking changes" and "not ship insecure software to users".

**Note: I have talked with upstream and they are indeed still planning to rebase this, it's just going very slowly, so there is potentially another way to work around the security issues until they have a new release.**

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by kloenk — APPROVED** [review]
LGTM, can be added later again if upstream releases new versions

**Review by sefidel — CHANGES_REQUESTED** [review]
Agree with the decision here, LGTM other than the typo on the aliases file.

**Comment by sefidel** [comment]
I've closed #267618 as it was a hacky solution with lingering wayland problems.

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -841,6 +841,8 @@ mapAliases ({
   s2n = s2n-tls; # Added 2021-03-03
   sane-backends-git = sane-backends; # Added 2021-02-19
   scantailor = scantailor-advanced; # Added 2022-05-26
+  schildichat-web = throw "'schildichat-web' has been removed as it is unmaintained and does not receive security fixes.\nPlease participate in upstream discussion on getting out new releases:\nhttps://github.com/SchildiChat/schildichat-desktop/issues/212\nhttps://github.com/SchildiChat/schildichat-desktop/issues/215"; # ADded 2023-12-05
+  schildichat-desktop = throw "'schildichat-desktop' has been removed as it is unmaintained and does not receive security fixes.\nPlease participate in upstream discussion on getting out new releases:\nhttps://github.com/SchildiChat/schildichat-desktop/issues/212\nhttps://github.com/SchildiChat/schildichat-desktop/issues/215"; # Added 2023-12-05
```
Comment (sivizius) [inline pkgs/top-level/aliases.nix:845]: Maybe:

```nix
inherit (lib.genAttrs 
  [ "schildichat-web" "schildichat-desktop" ] 
  (name: throw '''
    '${name}' has been removed as it is unmaintained and does not receive security fixes.
    Please participate in upstream discussion on getting out new releases:
    https://github.com/SchildiChat/schildichat-desktop/issues/212
    https://github.com/SchildiChat/schildichat-desktop/issues/215''')
  ) schildichat-web schildichat-desktop; # Added 2023-12-05
```

or less dirty:

```nix
} // lib.genAttrs 
  [ "schildichat-web" "schildichat-desktop" ] 
  (name: throw '''
    '${name}' has been removed as it is unmaintained and does not receive security fixes.
    Please participate in upstream discussion on getting out new releases:
    https://github.com/SchildiChat/schildichat-desktop/issues/212
    https://github.com/SchildiChat/schildichat-desktop/issues/215''') # Added 2023-12-05
// {

**Review by yayayayaka — APPROVED** [review]
Agree with the decision, LGTM and willing to boop the merge button.

**Comment by TheArcaneBrony** [comment]
Hi, I just wanted to drop in that SchildiChat works fine if you override the package like this:
```
(
  schildichat-desktop.override {
    electron = pkgs.electron;
  }
)
```

Could this be seen as an alternative solution?

**Comment by Kloenk** [comment]
> Hi, I just wanted to drop in that SchildiChat works fine if you override the package like this:
> 
> ```
> (
>   schildichat-desktop.override {
>     electron = pkgs.electron;
>   }
> )
> ```
> 
> Could this be seen as an alternative solution?

This would still leave you with `CVE-2023-37259`

**Pull request merged**